mungegithub/mungers/approvers/owners.go

```diff
@@ -0,0 +1,200 @@
+	repo  RepoInterface
+	alias features.Aliases
+}
+
+func NewRepoAlias(repo RepoInterface, alias features.Aliases) RepoAlias {
+	return RepoAlias{
+		repo:  repo,
+		alias: alias,
+	}
+}
+
+func (r RepoAlias) Approvers(path string) sets.String {
+	return r.alias.Expand(r.repo.Approvers(path))
+}
+
+func (r RepoAlias) LeafApprovers(path string) sets.String {
+	return r.alias.Expand(r.repo.LeafApprovers(path))
+}
+
+func (r RepoAlias) FindOwnersForPath(path string) string {
+	return r.repo.FindOwnersForPath(path)
+}
+
+type Owners struct {
+	filenames []string
+	repo RepoInterface
+	seed int64
+}
+
+func (o Owners) NewOwners(filenames[]string, r RepoInterface, s int64) Owners {
+	return Owners{filenames: filenames, repo: r, seed: s}
+}
+
+
+// removeSubdirs takes a list of directories as an input and returns a set of directories with all
+// subdirectories removed.  E.g. [/a,/a/b/c,/d/e,/d/e/f] -> [/a, /d/e]
+func removeSubdirs(dirList []string) sets.String {
+	toDel := sets.String{}
+	for i := 0; i < len(dirList)-1; i++ {
+		for j := i + 1; j < len(dirList); j++ {
+			// ex /a/b has prefix /a so if remove /a/b since its already covered
+			if strings.HasPrefix(dirList[i], dirList[j]) {
+				toDel.Insert(dirList[i])
+			} else if strings.HasPrefix(dirList[j], dirList[i]) {
+				toDel.Insert(dirList[j])
+			}
+		}
+	}
+	finalSet := sets.NewString(dirList...)
+	finalSet.Delete(toDel.List()...)
+	return finalSet
+}
+
+func (o Owners) GetApprovers() map[string]sets.String {
+	ownersToApprovers := map[string]sets.String{}
+
+	for _, fn := range o.ListOwnersFiles() {
+		ownersToApprovers[fn] = o.repo.LeafApprovers(fn)
+	}
+
+	return ownersToApprovers
+}
+
+func (o Owners) GetReverseMap() map[string]sets.String {
+	approverOwnersfiles := map[string]sets.String{}
+	for ownersFile, approvers := range o.GetApprovers() {
+		for approver := range approvers {
+			if _, ok := approverOwnersfiles[approver]; ok {
+				approverOwnersfiles[approver].Insert(ownersFile)
+			} else {
+				approverOwnersfiles[approver] = sets.NewString(ownersFile)
+			}
+		}
+	}
+	return approverOwnersfiles
+}
+
+func (o Owners) GetShuffledApprovers() (people []string, files []sets.String) {
+	approverOwnersfiles := o.GetReverseMap()
+	sliceOfKeys := make([]string, 0, len(approverOwnersfiles))
+	for approver := range approverOwnersfiles {
+		sliceOfKeys = append(sliceOfKeys, approver)
+	}
+	sort.Strings(sliceOfKeys)
+	order := rand.Perm(len(approverOwnersfiles))
+	people = make([]string, 0, len(approverOwnersfiles))
+	for i := range sliceOfKeys {
+		people = append(people, sliceOfKeys[order[i]])
+	}
+
+	files = []sets.String{}
+	for _, person := range people {
+		files = append(files, approverOwnersfiles[person])
+	}
+	return people, files
+}
+
+
+func (o Owners) GetSuggestedApprovers() []string {
+
+	randomizedApprovers, filesCanApprove := o.GetShuffledApprovers()
+
+	copyOfFiles := sets.NewString()
+	for _, fn := range o.filenames {
+		copyOfFiles.Insert(fn)
+	}
```

Comment:
This could be replaced by ListOwnersFiles (especially if we make it return a set)